### PR TITLE
John's patch to stop souper from deleting instructions (which might h…

### PR DESCRIPTION
**Pull request by jubitaneja**
…ave side effects

or be terminators)

### CODE REVIEW

File: test/Pass/single-pass.ll [hunk 1]
```diff
@@ -6,7 +6,7 @@
 define i32 @foo(i32 %x) {
 entry:
   %add = add nsw i32 %x, 1
-  ; CHECK-NOT: icmp
+  ; CHECK: icmp
```
Comment (regehr) [inline test/Pass/single-pass.ll]: with this change you're looking for an icmp but an icmp already existed in the test case, so the test becomes useless. the point of this test is to look for an optimization happening. it's no longer shown by eliminating the icmp, you need to look at something different.

Comment (jubitaneja) [reply test/Pass/single-pass.ll]: I have fixed it. It tests the current functionality that instruction is not deleted but the uses are replaced by value, so we don't look for the uses of instruction now.

Comment (regehr) [reply test/Pass/single-pass.ll]: I think it'd be better to check for the whole line we want:
%conv = zext i1 true to i32
(or whatever it really is)

File: test/Pass/single-pass.ll [hunk 2]
```diff
@@ -6,8 +6,9 @@
...
```
Comment (regehr) [inline test/Pass/single-pass.ll]: also delete this CHECK

**Comment by jubitaneja** [comment]
Fixed.

**Comment by regehr** [comment]
Looks good to me -- do you think this makes sense, Raimondas or Yang?

**Comment by rsas** [comment]
LGTM, but better @chenyang78 should look into this as well.

**Comment by chenyang78** [comment]
Yes, LGTM, too. Thanks.

**Pull request merged**